stellargraph/layer/gcn.py

```diff
@@ -57,46 +57,45 @@ class GraphConvolution(Layer):
         use_bias (bool): toggles an optional bias
         final_layer (bool): If False the layer returns output for all nodes,
                             if True it returns the subset specified by the indices passed to it.
-        kernel_initializer (str or func): The initialiser to use for the weights;
-            defaults to 'glorot_uniform'.
-        kernel_regularizer (str or func): The regulariser to use for the weights;
-            defaults to None.
-        kernel_constraint (str or func): The constraint to use for the weights;
-            defaults to None.
-        bias_initializer (str or func): The initialiser to use for the bias;
-            defaults to 'zeros'.
-        bias_regularizer (str or func): The regulariser to use for the bias;
-            defaults to None.
-        bias_constraint (str or func): The constraint to use for the bias;
-            defaults to None.
+        kernel_initializer (str or func, optional): The initialiser to use for the weights.
+        kernel_regularizer (str or func, optional): The regulariser to use for the weights.
+        kernel_constraint (str or func, optional): The constraint to use for the weights.
+        bias_initializer (str or func, optional): The initialiser to use for the bias.
+        bias_regularizer (str or func, optional): The regulariser to use for the bias.
+        bias_constraint (str or func, optional): The constraint to use for the bias.
     """
 
     def __init__(
-        self, units, activation=None, use_bias=True, final_layer=False, **kwargs
+        self,
+        units,
+        activation=None,
+        use_bias=True,
+        final_layer=False,
+        input_dim=None,
+        kernel_initializer="glorot_uniform",
+        kernel_regularizer=None,
+        kernel_constraint=None,
+        bias_initializer="zeros",
+        bias_regularizer=None,
+        bias_constraint=None,
+        **kwargs
     ):
-        if "input_shape" not in kwargs and "input_dim" in kwargs:
-            kwargs["input_shape"] = (kwargs.get("input_dim"),)
+        if "input_shape" not in kwargs and input_dim is not None:
```

Comment:
Why does `input_shape` get special treatment and is still part of kwargs? Is it purely because of this parameter that we can't remove `**kwargs` from the argument list altogether?

Reply:
No, `input_shape` is a kwarg to Keras's `Layer` (https://github.com/tensorflow/tensorflow/blob/a641fa1c5c10f0a278c7671fd6f7df550a74935d/tensorflow/python/keras/engine/base_layer.py#L278-L291
) (along with other kwargs), and it's typical for Keras layer subclasses to have a `**kwargs` to pass through this and anything else, e.g.: https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense 's `__init__` has a `**kwargs` too.

Most of the stellargraph layers seem to have an `input_dim` argument to make defining single-dimensional input slightly easier, I guess.